recipes/hipace/test.bat

```diff
@@ -0,0 +1,12 @@
+@echo on
+
+set "OMP_NUM_THREADS=2"
+set "TEST_DIR=example\beam_in_vacuum"
+
+:: executable
+impactx.NOMPI.NOACC.DP.exe %TEST_DIR%\inputs_SI
+if errorlevel 1 exit 1
+
+:: ctest
+ctest --test-dir build --build-config RelWithDebInfo --output-on-failure
+if errorlevel 1 exit 1
```

Comment:
I will skip this because @isuruf told me on Gitter:
> adding build/ would work, but keep in mind that whatever is in source_files would end up in the final conda package in order to facilitate testing in user machines. (They won't get extracted to the user environment, but will be in the .conda file

Reply:
Reworked - using a subset of the test scripts now to validate correctness in the test phase (w/o `ctest`).